src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4585,7 +4586,7 @@ internal string FixupFileName(string moduleBase, string name, string extension,
             // Return the path if successfully resolved.
             if (resolvedPath != null)
             {
-                if (isImportingModule && resolvedPath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+                if (isImportingModule && resolvedPath.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) && !skipLoading)
```

Comment:
Why do we need `skipLoading` if we have `isImportingModule`?

Reply:
Even though we are importing the module, we should not be loading the files in FileList here. FileList is meant to be just a list of all the files in the module. It should not have effect on the loading.

Reply:
The new parameter description and name could be more informative.
`getFileListOnly`?